### DISCUSSION TITLE
Timeout Problem with Raspberry Pi 5 USB Port Output

### DISCUSSION THREAD

**Discussion by Mickey70** [pyusb/pyusb#508] [Q&A] [answered]
Everything looks correct for USB port configuration, but when I try to write any amount of data out of the RP5's USB port, I always get a time-out error. Any thoughts most welcome!

More background. I wrote a simple Arduino script to have the Mega2560 echo whatever ASCII I sent to it via a USB connection between my PC and the Arduino. On the PC, I used Putty over a serial connection to type in the characters I wanted to send. The echo worked fine. I next took the same Mega2560 and connected it to my RP5 using the same USB cable. As stated in the first line above, writing out on the RP5's USB port ALWAYS times out.

**Comment by mcuee** [comment]
This seems to have nothing to do with pyusb.

You should use serial connection and pyserial is a good one to use if you prefer to use Python.

**Comment by Mickey70** [comment, marked as answer]
I posted a relevant answer in my comments to #507 which could be helpful here.
